src/models/proof_abstractions/tasm/prover_job.rs

```diff
@@ -326,7 +328,13 @@ impl ProverJob {
                 let output = result?;
                 match output.status.code() {
                     Some(0) => {
-                        let proof = bincode::deserialize(&output.stdout)?;
+                        let proof: Proof = bincode::deserialize(&output.stdout)?;
+                        info!(
+                            "Generated proof, with padded height: {}",
+                            proof.padded_height()
+                                .map(|x| x.to_string())
+                                .unwrap_or("could not get padded height from proof.".to_string())
```

Comment:
style: let's prefer .unwrap_or_else(), eg:

```suggestion
                                .unwrap_or_else(|| "could not get padded height from proof.".to_string())
```

That way the code in the closure does not execute for the happy path.